**Задача 1**

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания mu с надёжностью 0.95 (доверительная информация), если выборочное среднее равно 80, а объём равен 256.



$\sigma = 16$
<br>
$n = 256$
<br>
$\bar{X} = 80$
<br>
Доверительный интервал для среднего с известной дисперсией ищем по формуле
<br>
$
\left( \overline{X} - t_{1 - \alpha/2, \: n - 1} \cdot \dfrac{\sigma_X}{\sqrt{n}} 
\leq \mu \leq
\overline{X} + t_{1 - \alpha/2, \: n - 1} \cdot \dfrac{\sigma_X}{\sqrt{n}} 
\right)
$
<br>
$\left( 80 - 2 \cdot \frac{16}{\sqrt{256}}; 80 + 2 \cdot \frac{16}{\sqrt{256}}\right) = (78;82)$


**Задача 2**

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

202, 203, 199, 197, 195, 201, 200, 204, 194, 190

Известно, что их веса распределены нормально.
1. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?
2. Найдите P-значение для данного теста.



$\bar{X} = 198,5$
<br>
$n = 10$
<br>
Ищем исправленное кв. отклонение
$\sigma = \sqrt{\frac{(x_i - \bar{x})^2}{n-1}} = \sqrt{\frac{178.5}{9}} = 4.453$
<br>
Предположим, что нулевая гипотеза равна:
<br>
$H_o: a = 200$ и противоположная гепотеза $H_1: a \neq 200$
<br>
Ишем T набл.
$T_{набл} = \frac{\bar{X} - a}{\sigma} \cdot \sqrt{n} = \frac{198, 5 - 200}{4,453}\cdot \sqrt{10} = -1,075$
Ишем по табличке критерий Стьюдента = $\alpha = 0,05;$ $n-1 = 9 $ 
<br>
$T_{кр} = 1,833$
<br>
Смотрим доверительный интервал:
<br>
$\left(\bar{X} - T_{kp} \cdot \frac{\sigma}{\sqrt{n}}; \bar{X} + T_{kp} \cdot \frac{\sigma}{\sqrt{n}}\right) = \left(198,5 - 1,833 \cdot\frac{4,453}{\sqrt{10}};198,5 + 1,833 \cdot\frac{4,453}{\sqrt{10}}\right) = (195,92; 201,08)$
<br>
Гипотезе $H_{0}$ можно верить, 200 входит в доверительный интервал. 

**Задача 3**
<br>
Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.



Гипотеза $H_{0}: Ø = 17$
<br>
Гипотеза $H_{1}: Ø > 17 $

Шариков много, значит распределение будет нормальное
Среднее значение выборки тож будет 17

$a = 17$
<br>
$n = 100$
<br>
$\bar{X} = 17,5$
<br>
$D = 4$
<br>
$\alpha = 0.05$
<br>
$\sigma = \sqrt{\dfrac{D(X)}{n}} = \sqrt {\dfrac{4}{100}} = 0.2$
<br>
Ищем критическую область
<br>
$\Omega_\alpha = (t_{1 - \alpha}, \infty)$
<br>
Используем правило 2 сигм
<br>
$(a - 2 \cdot \sigma,  a + 2 \cdot \sigma)$
<br>
$(17 - 2 * 0.2, 17 + 2 * 0,2) = (16.6;17,4)$
<br>
Считаем T(X)


In [8]:
from scipy import stats
import numpy as np
T_x = np.round(stats.norm.ppf(1 - 0.025, 17, 0.2), 4)
print(T_x)

17.392


17.39 попадает в интервал (16.6;17,4), гипотеза $H_{0}$ верна

**Задача 4 (Дополнительно)**

Дана выборка диаметров подшипников из примера 1 с занятия 5:

samples = [0.6603, 0.9466, 0.5968, 1.3792, 1.5481, 0.7515, 1.0681, 1.1134,
1.2088, 1.701 , 1.0282, 1.3579, 1.0191, 1.1784, 1.1168, 1.1372,
0.7273, 1.3958, 0.8665, 1.5112, 1.161 , 1.0232, 1.0865, 1.02 ]

Предполагая, что диаметры подшипников распределены нормально, проверьте гипотезу о том, что дисперсия случайной величины равна 0.0625 при уровне значимости alpha = 0.05. Что для этого нужно знать:
1. Альтернативная гипотеза двухсторонняя.
2. Статистика для теста: H = (n - 1) * sample_variance / variance, где n - число элементов в выборке, sample_variance - несмещённая оценка дисперсии, variance - утверждаемая нулевой гипотезой дисперсия.
3. Эта статистика в предположении верности нулевой гипотезы имеет распределение хи-квадрат с параметром df = n - 1. Её квантили можно найти с помощью функции scipy.stats.chi2.ppf.